src/com/koxudaxi/poetry/PyAddExistingPoetryEnvPanel.kt

```diff
@@ -55,20 +58,32 @@ class PyAddExistingPoetryEnvPanel(private val project: Project?,
                 .panel
         add(formPanel, BorderLayout.NORTH)
         addInterpretersAsync(sdkComboBox) {
-            detectPoetryEnvs(module, existingSdks, context, project?.basePath ?: newProjectPath)
+            val existingSdkPaths = sdkHomes(existingSdks)
+            val moduleSdks = allModules(project).parallelStream().flatMap { module ->
+                val sdks = detectPoetryEnvs(module, existingSdkPaths, module.basePath)
+                        .filterNot { it.isAssociatedWithAnotherModule(module) }
+                sdks.forEach { sdkToModule.putIfAbsent(it.name, module) }
+                sdks.stream()
+            }.toList()
+            val rootSdks = detectPoetryEnvs(module, existingSdkPaths, project?.basePath ?: newProjectPath)
                     .filterNot { it.isAssociatedWithAnotherModule(module) }
+            val moduleSdkPaths = moduleSdks.map { it.name }.toSet()
+            val sdks = rootSdks.filterNot { moduleSdkPaths.contains(it.name) } + moduleSdks
+            sdks.sortedBy { it.name }
```

Comment:
Sorted alphabetically, which is understandable and searchable (alternative was by module name):
<img width="875" alt="Add_Python_Interpreter_and_Project_Structure" src="https://user-images.githubusercontent.com/527952/116505032-cbe89200-a887-11eb-8f78-980c95d2d803.png">